lib/src/Base/Optim/TNC.cxx

```diff
@@ -213,14 +213,21 @@ void TNC::run()
 
   setResultFromEvaluationHistory(evaluationInputHistory_, evaluationOutputHistory_);
   result_.setStatusMessage(tnc_rc_string[returnCode - TNC_MINRC]);
+  if ((returnCode != TNC_LOCALMINIMUM) && (returnCode != TNC_FCONVERGED) && (returnCode != TNC_XCONVERGED) && (returnCode != TNC_USERABORT))
+    result_.setStatus(OptimizationResult::FAILURE);
```

Comment:
```suggestion
  if (!(returnCode == TNC_LOCALMINIMUM) || (returnCode == TNC_FCONVERGED) || (returnCode == TNC_XCONVERGED) || (returnCode == TNC_USERABORT))
    result_.setStatus(OptimizationResult::FAILURE);
```
IMO it would be better to define as a failure the fact that the return code is not one of the expected codes.

Reply:
its never standard across lib, there are no expected codes definition possible ; we have to define our own codes (FAILURE/SUCCESS/TIMEOUT for now)

Reply:
I don't really understand what you are saying.  With my suggestion, if any unexpected code comes up, it is assumed to be a failure.

Reply:
the current code is:
```
if ((returnCode != TNC_LOCALMINIMUM) && (returnCode != TNC_FCONVERGED) && (returnCode != TNC_XCONVERGED) && (returnCode != TNC_USERABORT))
    result_.setStatus(OptimizationResult::FAILURE);
```
so thats what it already does

Reply:
Indeed, but I thought my version was more expressive. It is not very important, feel free to ignore, I have approved anyway.

Reply:
ok